### DISCUSSION TITLE
Any guide to achieve responsiveness?

### DISCUSSION THREAD

**Discussion by ArnavK-09** [kaplayjs/kaplay#463] [Q&A] [answered]
For both bigger & smaller screens

**Comment by mflerackers** [comment]
There is an optimization guide in the docs. But it really depends on what you're trying to achieve.

**Reply by ArnavK-09** [reply]
Unable to find a link, can you share?

I want to achieve  game ui responsiveness

**Reply by mflerackers** [reply]
Did you actually check the docs? It's right there when you open them https://kaplayjs.com/guides/optimization/

**Reply by ArnavK-09** [reply]
Like resizing objects on smaller screen and vice versa

**Reply by lajbel** [reply, marked as answer]
- `width()` and `height()` returns the current size of canvas
- `kaplay({ stretch: true, letterbox: true })` will create black bars to mantain aspect ratio between `width` and `height`
- `scale()` component provides a way to scale game objects as you want
- `onResize()` triggers when the window or canvas is resized

I think there can be a guide in the near future anyway, but this info will help you now!
